### DISCUSSION TITLE
Why BC_Sym_Plane in turb solvers do nothing, no flux across the boundary should be done.

### DISCUSSION THREAD

**Discussion by ShiheJia** [su2code/SU2#2169] [Q&A]
I found that BC_Sym_Plane do nothing for turb solvers. For scalar vars, the symmetry BC is different from the euler BC that only works on the normal gradient of the velocity. Maybe BC_Sym_Plane should take the form of CTransLMSolver::BC_HeatFlux_Wall, because the boundary conditions of the Langtry-Menter model is a symmetry condition.

'''
  inline void BC_Sym_Plane(CGeometry* geometry, CSolver** solver_container, CNumerics* conv_numerics,
                           CNumerics* visc_numerics, CConfig* config, unsigned short val_marker) override {
    /*--- Convective and viscous fluxes across symmetry plane are equal to zero. ---*/
  }
'''

'''
void CTransLMSolver::BC_HeatFlux_Wall(CGeometry *geometry, CSolver **solver_container, CNumerics *conv_numerics,
                                      CNumerics *visc_numerics, CConfig *config, unsigned short val_marker) {
...
}
'''

**Comment by sun5k** [comment]
Hi @ShiheJia  


The question seems to be that "It is necessary to solve the symmetry BC of the turbulence calculation with zero gradient as used in CTransLMSolver.cpp", is that right?
I guess the symmetry BC of the turbulence calculation has zero flux, so "Do noting for turbulent solver" it's for efficient calculate.

**Comment by ShiheJia** [comment]
Well, the problem lies on the BC_Sym_Plane for CScalarClass which is used for turb and transition model. It should contribute to zero normal flux but not it do nothing. CTransLMSolver::BC_HeatFlux_Wall is mentioned because it is actually a symmetry bc (fig 1), while SA and SST model has a value-given solidwall BC (fig 2).
Maybe we could achieve BC_Sym_Plane for CScalarClass just like the CTransLMSolver::BC_HeatFlux_Wall, it would not increase much cost.
![微信截图_20231117105435](https://github.com/su2code/SU2/assets/58663453/ef0b8bf2-6d50-480a-ad9e-5b0fc5ed592c)
![微信截图_20231117105449](https://github.com/su2code/SU2/assets/58663453/9a0b85f0-01f6-459c-b485-666ccfd88e84)

**Comment by ShiheJia** [comment]
Here is an example: a flat plate case, which contains sym bc when x<0. The k donot show a symmetry performance.
![flow](https://github.com/su2code/SU2/assets/58663453/a7191a54-a023-4970-ae49-521fccd67103)
![untitled](https://github.com/su2code/SU2/assets/58663453/82c8ba7c-236a-4af2-b580-edb5d10abef6)
The problem may have influence on those half model cases.

**Comment by ShiheJia** [comment]
OK, this problem seems to be a difficult historical legacy that has been discussed many times in #1168, #657, #1625, #1125.

**Comment by pcarruscag** [comment]
Fluid problems are more difficult because velocity is a vector quantity.
There are several approaches to deal with symmetry BCs, see the book by Blazek for example, one of the treatments mentioned there involved modifying the gradients to ensure there really is 0 normal gradient.
We don't do that in SU2 and thus probably get some contributions to the source term due to numerical issues on boundaries.
